### PR TITLE
Adds support for preserving some semver operators when updating

### PR DESCRIPTION
**Pull request by eamodio**
Also adds a new command for when a version satisfies the requirement but isn't the latest -- allowing for an update

### CODE REVIEW

**Comment by eamodio** [comment]
@pflannery rebased this back onto `master` and fixes all the conflict

File: src/providers/abstractCodeLensProvider.js [hunk 1]
```diff
@@ -84,11 +96,29 @@ export abstract class AbstractCodeLensProvider {
   }
 
   makeSatisfiedCommand(serverVersion, codeLensItem) {
-    codeLensItem.command = {
+   codeLensItem.command = {
```
Comment (pflannery) [inline src/providers/abstractCodeLensProvider.js]: @eamodio can you format the spacing here. Should be two spaces before `codeLensItem`

**Comment by pflannery** [comment]
This looks good minus the missing space issue. thanks

**Comment by eamodio** [comment]
@pflannery fixed

**Pull request merged**